internal/producer/src/validation/validation.go

```diff
@@ -70,3 +70,55 @@ func ValidateContract(c *contracts.Contract) error {
 
 	return errors.New("Failed to validate contract")
 }
+
+// ValidatePending validates a contract with the given pending balance and pending state nonce
+func ValidatePending(c *contracts.Contract, pBalance *uint64, pNonce *uint64) error {
```

Comment:
Just curious, why are the uint64's passed as pointers?

Reply:
I was thinking that rather than returning 0's and an non-nil error for an invalid contract, we can just update the pending balance and state nonce in the func when the validation is successful so that we can prevent any accidental updates to the actual pending balance and state nonce of a sender when the contract validation fails.